service/controller/resource/monitoring/pvresizingresource/create.go

```diff
@@ -0,0 +1,108 @@
+package pvresizingresource
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/microerror"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/giantswarm/prometheus-meta-operator/v2/pkg/pvresizing"
+	"github.com/giantswarm/prometheus-meta-operator/v2/service/key"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	r.logger.Debugf(ctx, "ensuring pv resizing")
+	{
+		cluster, err := key.ToCluster(obj)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		namespace := fmt.Sprintf("%s-%s", cluster.GetName()+"-prometheus")
+		pvc, err := r.fetchPVC(ctx, cluster.GetName(), namespace)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		currentValue := pvc.Spec.Resources.Requests.Storage().String()
+		anoVal := cluster.GetAnnotations()[key.PrometheusDiskSizeAnnotation]
+		desireDiskSize := pvresizing.PrometheusDiskSize(anoVal)
+
+		// Check the value of annotation with the current value in PVC.
+		if currentValue != desireDiskSize {
+			// if different we can trigger an update
+			// https://github.com/prometheus-operator/prometheus-operator/issues/4079#issuecomment-1211989005
+			err = r.updatePrometheusDiskSize(ctx, cluster.GetName(), namespace, desireDiskSize, pvc)
+			if err != nil {
+				return microerror.Mask(err)
+			}
+		}
+
+	}
+	r.logger.Debugf(ctx, "ensured pv resizing")
+
+	return nil
+}
+
+func (r *Resource) fetchPVC(ctx context.Context, clusterID, namespace string) (*corev1.PersistentVolumeClaim, error) {
```

Comment:
PVC are named prometheus-<cluster_id>-db-prometheus-<cluster_id>-0. I'm wondering if we should not use a Get instead.
If you'd rather use a list, I would actually return a list of PVC (in case we move to sharding later) than return the first object without checking if there are more than 1 in the slice

Reply:
I prefer to list PVC with the selector,